addOns/selenium/src/main/java/org/zaproxy/zap/extension/selenium/SeleniumOptions.java

```diff
@@ -334,4 +360,79 @@ public void setPhantomJsBinaryPath(String phantomJsBinaryPath) {
                     PHANTOM_JS_BINARY_KEY, PHANTOM_JS_BINARY_SYSTEM_PROPERTY, phantomJsBinaryPath);
         }
     }
+
+    public List<BrowserExtension> getEnabledBrowserExtensions(Browser browser) {
+        return this.getBrowserExtensions().stream()
+                .filter(BrowserExtension::isEnabled)
+                .filter(be -> be.getBrowser() == browser)
+                .collect(Collectors.toList());
+    }
+
+    public List<BrowserExtension> getBrowserExtensions() {
+        List<BrowserExtension> list = new ArrayList<>();
+        if (EXTENSIONS_DIR.exists() && EXTENSIONS_DIR.isDirectory()) {
+            // Always read these from filestore so we always pickup new files
+            for (File file : EXTENSIONS_DIR.listFiles()) {
+                if (BrowserExtension.isBrowserExtension(file)) {
+                    BrowserExtension ext = new BrowserExtension(file);
+                    ext.setEnabled(!this.disabledExtensions.contains(ext.getFile().getName()));
+                    list.add(ext);
+                }
+            }
+        }
+        return list;
+    }
+
+    public void setBrowserExtensions(List<BrowserExtension> exts) {
+        this.disabledExtensions.clear();
+        // Delete any that are not in the list
+        for (File file : EXTENSIONS_DIR.listFiles()) {
+            if (BrowserExtension.isBrowserExtension(file)) {
+                boolean found = false;
+                for (BrowserExtension ext : exts) {
+                    if (ext.getFile().equals(file)) {
+                        found = true;
+                        break;
+                    }
+                }
+                if (!found) {
+                    try {
+                        Files.delete(file.toPath());
+                    } catch (IOException e) {
+                        LOGGER.error(
+                                "Failed to delete browser addon {}", file.getAbsoluteFile(), e);
```

Comment:
Does it worth normalise on add-on rather extension? (Don't mind which one.)

Reply:
Current versions of Firefox do call them add-ons (with a hyphen, if we want to be consistent/specific). I think within the industry it's understood either way.

We had been trying to be specific and consistent for ZAP stuff (I believe) because add-ons can have extensions, but not all extensions are add-ons (or something like that :wink:).

Reply:
Arg. I had tried to be consistant and use "extensions" for the browser thingies on the basis that we already refer to ZAP "add-ons" in loads of places. 
Firefox does also use the term "extensions" as a subset of add-ons: https://addons.mozilla.org/en-GB/firefox/extensions/
On that basis this should be "extension" - everyone ok with that?

Reply:
(I've pushed that change, but can obviously change things again if necessary...)